src/components/boolean-search/boolean-table.tsx

```diff
@@ -0,0 +1,307 @@
+    ],
+    [rows]
+  );
+
+  const data: any = useMemo(
+    () =>
+      rows?.map((row: any, index: any) => ({
+        // Map your rows data here
+        and: index === 0 ? '' : 'AND',
+        strength: (
+          <select
+            value={row.strength || defaultValues.strength}
+            onChange={(e) => handleChange(index, 'strength', e.target.value)}
+          >
+            {strength?.map((option: any, index: any) => {
+              return <option key={index}>{option.Strength}</option>;
+            })}
+          </select>
+        ),
+        source: (
+          <>
+            <input
+              className='red-text'
+              value={row.source.name || ''}
+              disabled
+              required
+              onChange={(e) => handleChange(index, 'source', e.target.value)}
+            />
+            {row.sourceInvalid ? <div className='red-text'>Please enter a value for this field.</div> : <></>}
+          </>
+        ),
+        stages: (
+          <>
+            <span>From: </span>
+            <select
+              className='stageDropdown'
+              value={row.stageFrom.Name || defaultValues.stageFrom.Name}
+              onChange={(e) => handleChange(index, 'stageFrom', e.target.value)}
+            >
+              {stageFrom?.map((option: any, index: any) => {
+                return <option key={index}>{option.Name}</option>;
+              })}
+            </select>
+            <span>To: </span>
+            <select
+              className='stageDropdown'
+              value={row.stageTo.Name || defaultValues.stageTo.Name}
+              onChange={(e) => handleChange(index, 'stageTo', e.target.value)}
+            >
+              {row.toStageOptions?.map((option: any, index: any) => {
+                return <option key={index}>{option.Name}</option>;
+              })}
+            </select>
+          </>
+        ),
+        pattern: (
+          <select
+            value={row.pattern || defaultValues.pattern || ''}
+            onChange={(e) => handleChange(index, 'pattern', e.target.value)}
+          >
+            <option style={{ display: 'none' }} disabled selected value=''></option>
+            {pattern?.map((option: any, index: any) => (
+              <option key={index}>{option.Pattern}</option>
+            ))}
+          </select>
+        ),
+        location: (
+          <select
+            value={row.location || defaultValues.location || ''}
+            onChange={(e) => handleChange(index, 'location', e.target.value)}
+          >
+            <option style={{ display: 'none' }} disabled selected value=''></option>
+            {location?.map((option: any, index: any) => {
+              return (
+                <option key={index}>
+                  {option.Pattern_Location}
+                </option>
+              );
+            })}
+          </select>
+        ),
+        actions: <button onClick={(e) => handleRemoveRow(e, index)}>Remove</button>
```

Comment:
I don't think this "data"  row is being used. I don't see a button named "Remove" anywhere on the page. Where I would expect this to show, I see the contents of the `Cell` from the `columns` constant.